adp-algorithms/util/src/main/java/de/metanome/util/InclusionDependencyUtil.java

```diff
@@ -0,0 +1,86 @@
+package de.metanome.util;
+
+import com.google.common.base.Preconditions;
+import de.metanome.algorithm_integration.ColumnIdentifier;
+import de.metanome.algorithm_integration.results.InclusionDependency;
+import it.unimi.dsi.fastutil.ints.Int2IntMap;
+import it.unimi.dsi.fastutil.ints.Int2IntOpenHashMap;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.LinkedList;
+import java.util.List;
+
+public class InclusionDependencyUtil {
+
+  public List<InclusionDependency> getMax(final Collection<InclusionDependency> toMinify) {
```

Comment:
Should we make this static since it only depends on its parameter? That way we don't have to create new instances just to convert the input.

Reply:
True, but that also hinders testing. If the `getMax` method was to be stubbed, one would have a hard time. One could not just derive and override. Alternatively one would have to use a super-intrusive mocking framework for swapping out the impl. I like the fact that requiring an instance may require maintaining a field on the using class which makes all of its collaborators visible upfront.
For this class making the method static would mean adding collaborators is hard -  solutions include calling other static methods, or having a static field (and possibly static blocks to initialize), .... which are harder to maintain & test.